packages/trader/src/AppV2/Components/TradeParameters/Stake/stake.tsx

```diff
@@ -76,8 +76,11 @@ const Stake = observer(({ is_minimized }: TTradeParametersProps) => {
         has_error_1 && (error_field_1 === 'amount' || error_field_1 === 'stake') ? message_1 : '';
     const proposal_error_message_2 =
         has_error_2 && (error_field_2 === 'amount' || error_field_2 === 'stake') ? message_2 : '';
+    const has_both_errors = has_error_1 && has_error_2;
     const proposal_error_message =
-        proposal_error_message_1 || proposal_error_message_2 || validation_errors?.amount?.[0];
+        contract_types[1] && has_both_errors
```

Comment:
This became more and more complicated, Stake component is candidate N.1 for refactoring. Maybe we can split this condition to make it more readable? Something like this:

```
const show_proposal_error =
    contract_types[1] && has_both_errors;

const additional_proposal_error =
    !(contract_types[1] && !has_both_errors) && proposal_error_message_1;

const proposal_error_message = show_proposal_error
    ? proposal_error_message_1 || proposal_error_message_2 || validation_errors?.amount?.[0]
    : additional_proposal_error || validation_errors?.amount?.[0];
```